src/cunumeric/unary/scalar_unary_red_template.inl

```diff
@@ -145,19 +258,34 @@ static void scalar_unary_red_template(TaskContext& context)
   auto& inputs  = context.inputs();
   auto& scalars = context.scalars();
 
+  auto op_code     = scalars[0].value<UnaryRedCode>();
+  auto shape       = scalars[1].value<DomainPoint>();
+  bool has_where   = scalars[2].value<bool>();
+  size_t start_idx = has_where ? 2 : 1;
   std::vector<Store> extra_args;
-  for (size_t idx = 1; idx < inputs.size(); ++idx) extra_args.push_back(std::move(inputs[idx]));
-
-  auto op_code = scalars[0].value<UnaryRedCode>();
-  auto shape   = scalars[1].value<DomainPoint>();
+  for (size_t idx = start_idx; idx < inputs.size(); ++idx)
+    extra_args.push_back(std::move(inputs[idx]));
```

Comment:
consider reserving first
```suggestion
  extra_args.reserve(inputs.size());
  for (size_t idx = start_idx; idx < inputs.size(); ++idx)
     extra_args.emplace_back(std::move(inputs[idx]));
```
and/or also consider making this a range-for, you don't need the index:
```suggestion
  extra_args.reserve(inputs.size());
  for (auto& input : inputs) extra_args.emplace_back(std::move(input));
```
However, if you are feeling REALLY spicy, you can also just use the algorithm version of `std::move()` to achieve the same thing:
```suggestion
  extra_args.reserve(inputs.size());
  std::move(inputs.begin(), inputs.end(), std::back_inserter(extra_args));
```
----

But perhaps to a larger point, these are the `TaskContext` inputs. Are we allowed to move out of them?